cpp/src/io/text/multibyte_split.cu

```diff
@@ -166,7 +172,7 @@ __global__ void multibyte_split_kernel(
 
   // STEP 3: Flag matches
 
-  uint32_t thread_offsets[ITEMS_PER_THREAD];
+  int64_t thread_offsets[ITEMS_PER_THREAD];
```

Comment:
I wonder if a `#pragma unroll` for the for-loop below would improve performance?
Should the benchmarks for this be updated?

Reply:
@vuule may want to test this as part of #10325

Reply:
I can check if this affects perf, but no need to block the PR on this IMO.

Reply:
No perf impact from unroll when running the benchmark w/o GDS.